package/linux-firmware/linux-firmware.mk

```diff
@@ -835,13 +848,33 @@ LINUX_FIRMWARE_LICENSE_FILES = $(sort $(LINUX_FIRMWARE_ALL_LICENSE_FILES))
 # sure we canonicalize the pointed-to file, to cover the symlinks of the form
 # a/foo -> ../b/foo  where a/ (the directory where to put the symlink) does
 # not yet exist.
+#
+# If BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS is enabled, modules are compressed
+# in this stage after being copied to the target directory. The compression
+# is not done on modules in the images directory, because they must not be
+# compressed when bundled via the EXTRA_FIRMWARE option.
+#
+# $1: target directory
+# $2: 'images' (uncompressed) or 'target' (compressed if enabled)
 define LINUX_FIRMWARE_INSTALL_FW
 	mkdir -p $(1)
 	$(TAR) xf $(@D)/br-firmware.tar -C $(1)
+	if [ "$(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS)" = "y" ] && [ "$(2)" = "target" ]; then \
+		$(TAR) tf $(@D)/br-firmware.tar | while read f; do \
+			if [ -f $(1)/$$f ]; then \
+				$(LINUX_FIRMWARE_COMPRESS_CMD) $(1)/$$f; \
+			fi ; \
+		done ; \
+	fi
 	cd $(1) ; \
 	sed -r -e '/^Link: (.+) -> (.+)$$/!d; s//\1 \2/' $(@D)/WHENCE | \
 	while read f d; do \
-		if test -f $$(readlink -m $$(dirname "$$f")/$$d); then \
+		if [ "$(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS)" = "y" ] && [ "$(2)" = "target" ]; then \
```

Comment:
Could we in that case just store a suffix in a local variable (and that local variable is an empty string `""` by default)?

Reply:
With that we could move that if statement before the while...

Reply:
Adjusted per your comment (if I understood it correctly). Technically we don't need to check for `[ "$(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS)" = "y" ]` here because without it the `LINUX_FIRMWARE_COMPRESSED_FILE_SUFFIX` will be empty - but I think it's better to keep the check consistent.